client/src/components/Forums/ThreadDetails/ThreadDetails.tsx

```diff
@@ -59,6 +59,9 @@ const ThreadDetail = ({ forumId, threadId }: ThreadDetailProps) => {
     }
   };
 
+  const totalPosts: number = posts.length + 1;
+
+  if (pending) return null;
```

Comment:
the `pending` state seems to repeat the functionality of the `loading` state. Could we reuse the `loading` state in the handleDeletePost method or do we need the separate `pending` state?

Reply:
@seantokuzo I see what you are saying for sure. They are two different situations so I do feel there is value in differentiating though. It was really a quick fix to make sure there was some state-handling logic while it awaits migration to redux where the reducer state will handle all of this and not feel so clunky. I think we can leave it for now.